sky/spot/controller.py

```diff
@@ -175,15 +185,45 @@ def run(self):
             subprocess_utils.kill_children_processes()
             spot_state.set_cancelled(self._job_id)
         except exceptions.ResourcesUnavailableError as e:
+            # The exception will be caught when:
+            # 1. The strategy_executor fails to launch/recover the cluster
+            #    after the max number of retries when retry_until_up is not set.
+            # 2. None the failovers are caused by resource unavailability, i.e.
+            #    the job is configured incorrectly, e.g.,
+            #    InvalidClusterNameError, NotSupportedError,
+            #    CloudUserIdentityError, etc.
             logger.error(f'{common_utils.class_fullname(e.__class__)}: '
                          f'{colorama.Fore.RED}{e}{colorama.Style.RESET_ALL}')
-            spot_state.set_failed(
-                self._job_id,
-                failure_type=spot_state.SpotStatus.FAILED_NO_RESOURCE)
+            if any(
+                    isinstance(err, exceptions.ResourcesUnavailableError)
+                    for err in e.failover_history):
+                # If any of the failover fails due to resource unavailability,
+                # the spot job should be marked as FAILED_NO_RESOURCE, as the
+                # spot job may be able to launch next time.
+                spot_state.set_failed(
+                    self._job_id,
+                    failure_type=spot_state.SpotStatus.FAILED_NO_RESOURCE,
+                    failure_reason=str(e))
```

Comment:
It will be `f'Failed to launch the spot cluster after {max_retry} retries.'` from https://github.com/skypilot-org/skypilot/blob/78696c64b61bd58cb15a53ab4ff35aa5c6f9b4c6/sky/spot/recovery_strategy.py#L280-L283, as we don't break the retry if there is at least 1 `ResourcesUnavailableError` in the failover reasons.